dm/dm/ctl/master/task_create.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/pingcap/tiflow/dm/pkg/log"
+	"github.com/pingcap/tiflow/dm/pkg/terror"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+// NewTaskCreateCmd creates a Task Create command.
+func NewTaskCreateCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create <config-file>",
+		Short: "Create a task by task configuration file",
+		RunE:  taskCreateFunc,
+	}
+	return cmd
+}
+
+func taskCreateFunc(cmd *cobra.Command, _ []string) error {
+	if len(cmd.Flags().Args()) != 1 {
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+	res, err := createTask(cmd.Flags().Arg(0))
+	if err != nil {
+		return errors.Trace(err)
+	}
+	common.PrettyPrintOpenapiResp(true, res)
+	return nil
+}
+
+func createTask(filePath string) (*openapi.OperateTaskResponse, error) {
+	task, err := convertToOpenapiTaskFromFile(filePath)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// create task
+	params := []interface{}{openapi.CreateTaskRequest{Task: *task}}
+	createResp := &openapi.OperateTaskResponse{}
+	sendError := common.SendOpenapiRequest(ctx, "DMAPICreateTask", params, http.StatusCreated, createResp)
+	if sendError != nil {
+		return nil, errors.Trace(sendError)
+	}
+	return createResp, nil
+}
+
+func convertToOpenapiTaskFromFile(filePath string) (*openapi.Task, error) {
+	var (
+		err         error
+		sendError   *common.SendToOpenapiError
+		fileContent []byte
+		taskStr     string
+		params      []interface{}
+	)
+	fileContent, err = common.GetFileContent(filePath)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	task := config.NewTaskConfig()
+	err = task.RawDecode(string(fileContent))
+	if err != nil {
+		return nil, err
+	}
+	if task.TargetDB != nil && task.TargetDB.Security != nil {
+		loadErr := task.TargetDB.Security.LoadTLSContent()
+		if loadErr != nil {
+			log.L().Warn("load tls content failed", zap.Error(terror.ErrCtlLoadTLSCfg.Generate(loadErr)))
+		}
+		fileContent = []byte(task.String())
+	}
+
+	lines := bytes.Split(fileContent, []byte("\n"))
+	// we check if `is-sharding` is explicitly set, to distinguish between `false` from default value
+	isShardingSet := false
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("is-sharding")) {
+			isShardingSet = true
+			break
+		}
+	}
+	// we check if `shard-mode` is explicitly set, to distinguish between "" from default value
+	shardModeSet := false
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("shard-mode")) {
+			shardModeSet = true
+			break
+		}
+	}
+
+	if isShardingSet && !task.IsSharding && task.ShardMode != "" {
+		common.PrintLinesf("The behaviour of `is-sharding` and `shard-mode` is conflicting. `is-sharding` is deprecated, please use `shard-mode` only.")
+		return nil, errors.New("please check output to see error")
+	}
+	if shardModeSet && task.ShardMode == "" && task.IsSharding {
```

Comment:
Why is `shard-mode` set but `task.ShardMode` remains empty?

Reply:
the same as above.

Reply:
Should we pull out the same logic with `start-task`? Or it would be a bit difficult to make them aligned.

Reply:
I thought about it, too，but i'm afraid we may change the logic about check the input for extending commands and we may have to think about grpc interface, because previous commands still call this function.

Reply:
this logic has been pulled out in [convertToOpenapiTaskFromFile](https://github.com/pingcap/tiflow/pull/5901/files#diff-277a02d1cc0098a357a31a6eefddaa67508f1997d193cc928c21485a30a9767fR73) and it will be called by `task create`, `task start` or `task update` which will really use file content, not only task name,  `task check` will call `convertToOpenapiTaskFromFile` too, but it wasn't called by previous commands like `start-task` or `check-task`. i can pull out some logic and let it called by both previous commands and new commands, but if we need extending new commands, we may need think about previous too. considering that the previous instructions will be hidden, i am not sure if this is necessary.